### PR TITLE
Fix the build due to jackson-core incompatibility

### PR DESCRIPTION
**Pull request by jianglai**
This change is [Reviewable](https://reviewable.io/reviews/google/nomulus/2085)

### CODE REVIEW

File: java_common.gradle [hunk 1]
```diff
@@ -70,6 +70,26 @@ configurations {
...
 }
 
+// Custom resolution strategy to get around tricky dependency issues.
+configurations.all {
+  // jackson-core is a transitive dependency that we cannot explicity specify
+  // versions for, without triggering linter alerts. Even though the lockfiles
+  // lock it to a lower version, Gradle will first try to resolve the
+  // dependency tree based on what is available in the Maven repo before
+  // applying any dependency locks. The newer version (v2.15) of jackson-core
+  // results in a build failre for pre-7.6 Gradle versions during that stage.
```
Comment (CydeWeys) [inline java_common.gradle:80]: failure

File: java_common.gradle [hunk 2]
```diff
@@ -70,6 +70,26 @@ configurations {
   }
 }
 
+// Custom resolution strategy to get around tricky dependency issues.
+configurations.all {
```
Comment (CydeWeys) [inline java_common.gradle:74]: Add a TODO to remove this once we migrate to Gradle 7.6+ ?  (Or whatever the relevant condition is?)

**Review by jianglai — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/2085)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @CydeWeys and @weiminyu)

___
*[`java_common.gradle` line 74 at r1](https://reviewable.io/reviews/google/nomulus/2085#gh-1273629549:-NaCR3GW9brm_p2u4JK2:b-896fix) ([raw file](https://github.com/google/nomulus/blob/e6018e260f4927dff841138b681e6759e38e00f4/java_common.gradle#L74)):*
<details><summary><i>Previously, CydeWeys (Ben McIlwain) wrote…</i></summary><blockquote>

Add a TODO to remove this once we migrate to Gradle 7.6+ ?  (Or whatever the relevant condition is?)
</blockquote></details>

Done.
___
*[`java_common.gradle` line 80 at r1](https://reviewable.io/reviews/google/nomulus/2085#gh-1273628376:-NaCR32O8vV6x_FL8RAe:b-896fix) ([raw file](https://github.com/google/nomulus/blob/e6018e260f4927dff841138b681e6759e38e00f4/java_common.gradle#L80)):*
<details><summary><i>Previously, CydeWeys (Ben McIlwain) wrote…</i></summary><blockquote>

failure
</blockquote></details>

Done.

**Review by weiminyu — APPROVED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/2085)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @CydeWeys)

**Pull request merged**